app/src/main/java/com/techcourse/DispatcherServlet.java

```diff
@@ -1,28 +1,39 @@
 package com.techcourse;
 
-import com.interface21.webmvc.servlet.view.JspView;
+import com.interface21.webmvc.servlet.ModelAndView;
+import com.interface21.webmvc.servlet.View;
+import com.interface21.webmvc.servlet.mvc.tobe.AnnotationHandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.tobe.AnnotationHandlerMapping;
+import com.interface21.webmvc.servlet.mvc.tobe.HandlerAdapter;
+import com.interface21.webmvc.servlet.mvc.tobe.HandlerMapping;
+import com.interface21.webmvc.servlet.mvc.tobe.ManualHandlerAdapter;
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServlet;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
-import java.util.Collections;
+import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class DispatcherServlet extends HttpServlet {
 
     private static final long serialVersionUID = 1L;
+    private static final String basePackage = "interface21";
```

Comment:
```suggestion
    private static final String basePackage = "techcourse.controller";
```

basePackage 의 설정 값이 수정되어야 합니다!

어노테이션 컨트롤러를 작성한다면, `app/src/main/java/com/techcourse/controller` 하위에 작성하게 될 것 같은데요. 지금 코드에서는 basePackage가 mvc 프레임워크 영역에 해당하는 `interface21`로 설정되어 있어서 **`app/` 하위의 controller를 찾을 수 없습니다.**

이 부분을 수정 한 다음에 `com.techcourse.controller`에 아래 테스트 컨트롤러 추가 후, `http://localhost:8080/test` 위치로 접근해서 `index.jsp`를 볼 수 있는지 확인해보세요!

```java
@Controller
public class Step2Controller {

    @RequestMapping(value = "/test", method = RequestMethod.GET)
    public ModelAndView showForm(HttpServletRequest req, HttpServletResponse res) {
        final var modelAndView = new ModelAndView(new JspView("/index.jsp"));
        return modelAndView;
    }
}
```

Reply:
스캔 범위 지정을 했어야 했는데 못했네요!! 꼼꼼하게 찾아주셔서 감사해요 노랑! 테스트 컨트롤러까지 같이 주셔서 감사해요. 수정했습니다!